### ISSUE TITLE
Allow post urls to be generated based on post metadata

### ISSUE THREAD

**Issue by bebraw** [antwarjs/antwar#4]
Example:

```javascript
// basic
function basicUrlify(post) {
    return post.category + '/' + slugify(post.title);
}

// multiple urls per post
function multipleUrlify(post) {
    return [
       post.category + '/' + slugify(post.title),
       dateify(post.date) + '/' + slugify(post.title)
    ];
}

// redirect
function redirectUrlify(post) {
    var to = dateify(post.date) + '/' + slugify(post.title);

    return [
        {
            from: post.category + '/' + slugify(post.title),
            to: to,
        },
        to,
    ];
}
```

You would just have one of these mappings somewhere in the project configuration. The nice thing is that it's very flexible and you can do 1-1, N-1 type of mappings and even rerouting (last case).

**Comment by eldh** [comment]
Yes! Great idea!

**Comment by eldh** [comment]
I'll start with a basic example in the config.

**Comment by eldh** [comment]
Fixed in 180faf8.
